plutus-tx-plugin/test/Budget/filter.plc.golden

```diff
@@ -75,102 +75,118 @@
             )
           )
         )
-        [
-          [
+        (let
+          (nonrec)
+          (termbind
+            (strict)
+            (vardecl eta [ List (con integer) ])
             [
-              { { foldr (con integer) } [ List (con integer) ] }
-              (lam
-                e
-                (con integer)
-                (lam
-                  xs
-                  [ List (con integer) ]
-                  {
-                    [
+              [
+                (let
+                  (nonrec)
+                  (typebind (tyvardecl a (type)) [ List (con integer) ])
+                  (lam
+                    c
+                    (fun (con integer) (fun a a))
+                    (lam
+                      n
+                      a
                       [
-                        {
+                        [ c (con integer 1) ]
+                        [
+                          [ c (con integer 2) ]
                           [
-                            Bool_match
+                            [ c (con integer 3) ]
                             [
+                              [ c (con integer 4) ]
                               [
+                                [ c (con integer 5) ]
                                 [
-                                  { (builtin ifThenElse) Bool }
+                                  [ c (con integer 6) ]
                                   [
+                                    [ c (con integer 7) ]
                                     [
-                                      (builtin equalsInteger)
+                                      [ c (con integer 8) ]
                                       [
-                                        [ (builtin modInteger) e ]
-                                        (con integer 2)
+                                        [ c (con integer 9) ]
+                                        [ [ c (con integer 10) ] n ]
                                       ]
                                     ]
-                                    (con integer 0)
                                   ]
                                 ]
-                                True
                               ]
-                              False
                             ]
                           ]
-                          (all dead (type) [ List (con integer) ])
-                        }
-                        (abs dead (type) [ [ { Cons (con integer) } e ] xs ])
+                        ]
                       ]
-                      (abs dead (type) xs)
-                    ]
-                    (all dead (type) dead)
-                  }
+                    )
+                  )
                 )
-              )
+                (lam
```

Comment:
Same eta-reduction problem as `monoConstructor.plc.golden`. Other than this, the old and the new versions are more or less equivalent, just that things appear in slightly different orders.